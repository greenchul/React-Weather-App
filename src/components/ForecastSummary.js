import React from "react";
import propTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import moment from "moment";

const ForecastSummary = ({ singleForecast, onSelect }) => {
  const { date, temperature, description, icon } = singleForecast;
  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max} &deg;C
      </div>
      <div className="forecast-summary__description">{description}</div>

      <button
        type="button"
        className="forecast-summary__button"
        onClick={() => onSelect(date)}
      >
        More details
      </button>
    </div>
  );
};

ForecastSummary.propTypes = {
  singleForecast: propTypes.shape({
    date: propTypes.number.isRequired,
    temperature: propTypes.shape({
      max: propTypes.number,
      min: propTypes.number,
    }).isRequired,
    description: propTypes.string.isRequired,
    icon: propTypes.string.isRequired,
  }).isRequired,
  onSelect: propTypes.func.isRequired,
};

export default ForecastSummary;
