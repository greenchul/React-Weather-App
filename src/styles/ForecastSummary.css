.forecast-summary__grid-container {
  display: grid;
  grid-template-columns: 3fr 1fr;
}

.forecast-summary__temperature {
  font-size: 30px;
  font-weight: 1000;
}

.forecast-summary__grid-1 {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  font-size: 20px;
  margin-bottom: 5px;
}

.forecast-summary__grid-2 {
  display: flex;
  justify-content: center;
  align-items: center;
}

@media screen and (min-width: 768px) {
  .forecast-summary__temperature {
    font-size: 40px;
  }
  .forecast-summary__grid-1 {
    font-size: 25px;
    letter-spacing: 3px;
    line-height: 35px;
  }
  button {
    font-size: 20px;
  }
}

@media screen and (min-width: 1024px) {
}
