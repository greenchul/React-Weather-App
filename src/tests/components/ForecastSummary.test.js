import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("LocationDetails", () => {
  const singleForecast = {
    date: 12122022,
    icon: "1234",
    temperature: {
      max: 23,
      min: 18,
    },
    description: "Cloudy",
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary singleForecast={singleForecast} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct forecast data and props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary singleForecast={singleForecast} />
    );
    expect(getByText("Description: Cloudy")).toBeTruthy();
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getByText("Max temp: 23")).toBeTruthy();
    expect(getByText("Description: Cloudy")).toHaveClass(
      "forecast-summary__description"
    );
  });
});
