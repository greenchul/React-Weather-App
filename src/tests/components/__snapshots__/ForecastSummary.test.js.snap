// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
       Date: 12122022 
    </div>
    <div
      class="forecast-summary__temperature"
    >
      Max temp: 23
    </div>
    <div
      class="forecast-summary__description"
    >
      Description: Cloudy
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      Icon: 1234
    </div>
  </div>
</DocumentFragment>
`;
